py_neuromodulation/nm_stream.py

```diff
@@ -134,23 +139,37 @@ def _run(
                 data_batch.astype(np.float64)
             )
             if is_stream_lsl:
-                feature_dict["time"] = time_[-1]
                 if self.verbose:
                     if last_time is not None:
                         logger.debug("%.3f seconds of new data processed", time_[-1] - last_time)
+                        feature_dict["time"] = time_[-1] - last_time
+                    else:
+                        feature_dict["time"] = 0
                     last_time = time_[-1]
             else:
-                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 ).astype(int)
+                feature_dict["time"] = np.ceil(time_[-1] * 1000 +1 )
                 logger.info("Time: %.2f", feature_dict["time"]/1000)
             
 
             self._add_target(feature_dict, data_batch)
+            buff_cnt += 1
 
-            l_features.append(feature_dict)
+            feature_dict = db.cast_values(feature_dict)
+            if not data_acquired:
+                db.create_table(feature_dict)
+                data_acquired = True
+            db.insert_data(feature_dict)
 
-        feature_df = pd.DataFrame(l_features)
+            if buff_cnt >= 10:
```

Comment:
Please make the number of samples a function parameter. It would be even better if it's actually a time. So save data each 1 sec, 10 sec etc.

Reply:
So I added it as a parameter. I can easily change it to seconds instead of number of samples but I actually think number of feature samples would be more intuitive as users will know how many datapoints are possibly not written in the db in case of e.g. a crash. With seconds that could differ depending on the sampling rate, right? Also, when running a stream with a raw_data_generator time could be counter intuitive as there is the "real time" and the stream time (most likely running way faster). What do you think?

Reply:
Agree